tracer/test/test-applications/integrations/Samples.DataStreams.RabbitMQ/Program.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Text;
+using Datadog.Trace;
+using RabbitMQ.Client;
+
+namespace Samples.DataStreams.RabbitMQ
+{
+    public static class Program
+    {
+        private static readonly string Queue = nameof(Queue);
+        private static readonly string Exchange = nameof(Exchange);
+        private static readonly string RoutingKey = nameof(RoutingKey);
+        private static readonly string Message = nameof(Message);
+        private static readonly string Host = Environment.GetEnvironmentVariable("RABBITMQ_HOST") ?? "localhost";
+        
+        public static void Main(string[] args)
+        {
+            using (Tracer.Instance.StartActive("rabbitmq-sample"))
+            {
+                var factory = new ConnectionFactory() { HostName = Host };
+                using var connection = factory.CreateConnection();
+                using var model = connection.CreateModel();
+            
+                // produce/consume operation (direct exchange)
+                PublishMessageToQueue(model, Queue, Message);
+                var msg = GetMessage(model, Queue);
+
+                // continuing the chain
+                var queueName = PublishMessageToExchange(model, Exchange, RoutingKey, msg);
+                GetMessage(model, queueName);              
```

Comment:
Not requesting any changes per se, but do you know what happens if this GetMessage were to be called on another thread? Would the resulting datastreams points (and their pathways) be identical?

Reply:
It will continue the chain, as the context will be extracted from the consumed message headers